test/dynamo/test_skip_non_tensor.py

```diff
@@ -7,6 +7,59 @@
 import torch._dynamo.test_case
 from torch._dynamo.testing import CompileCounter
 
+_variable = 0
+_variable_2 = 0
+
+
+def user_function():
+    return torch._utils.is_compiling()
+
+
+def user_generator():
+    for _ in range(1):
+        yield torch._utils.is_compiling()
+    return
+
+
+class MyModule(torch.nn.Module):
+    def __init__(self, mode: int):
+        super().__init__()
+        self.mode = mode
+        self.register_forward_pre_hook(self.pre_forward, with_kwargs=True)
+
+    def pre_forward(self, module, args, kwargs):
+        if self.mode == 0:
+            if user_function():
+                global _variable
+                _variable += 1
+        return args, kwargs
+
+    def forward(self, x):
+        global _variable
+        global _variable_2
+        # There may be side-effects inconsistent with eager when
+        # compiling, this will force dynamo to commit the graph
+        if self.mode == 0:
+            # modify the variable
+            x += 1
+        elif self.mode == 1:
+            if torch._utils.is_compiling():
+                _variable += 1
+            else:
+                _variable_2 += 1
+        elif self.mode == 2:
+            if user_function():
+                _variable += 1
+        elif self.mode == 3:
+            lambda_f = lambda: torch._utils.is_compiling()  # noqa: E731
+            if lambda_f():
+                _variable += 1
+        elif self.mode == 4:
```

Comment:
Could you also test the case where there is a graph break under the is_compiling control flow?

Reply:
actually graph break would trigger compilation, so it will also commit the graph.

Reply:
anw, let's test it anyway, since it is expected behaviour to capture the control flow regardless of graph break